projects/plugins/crm/src/automation/class-automation-workflow.php

```diff
@@ -42,28 +42,27 @@ class Automation_Workflow {
 	/**
 	 * Automation_Workflow constructor.
 	 *
-	 * @param array             $workflow_data The workflow data to be constructed.
-	 * @param Automation_Engine $automation_engine An instance of the Automation_Engine class.
+	 * @param array $workflow_data The workflow data to be constructed.
 	 */
-	public function __construct( array $workflow_data, Automation_Engine $automation_engine ) {
+	public function __construct( array $workflow_data ) {
 		$this->id           = $workflow_data['id'] ?? null;
 		$this->triggers     = $workflow_data['triggers'] ?? array();
 		$this->initial_step = $workflow_data['initial_step'] ?? array();
 		$this->name         = $workflow_data['name'];
 		$this->description  = $workflow_data['description'] ?? '';
 		$this->category     = $workflow_data['category'] ?? '';
 		$this->active       = $workflow_data['is_active'] ?? true;
-
-		$this->automation_engine = $automation_engine;
-		$this->logger            = $automation_engine->get_logger() ?? Automation_Logger::instance();
 	}
 
 	/**
 	 * Get the id of this workflow
 	 *
-	 * @return int
+	 * This will either be a string if the workflow is registered in the codebase,
+	 * or an integer if it is a custom workflow stored in the database.
```

Comment:
OK. The id was added for the database row. Normally I try to follow the strict type way and avoid mixed types. Do you have a use case where we need the ID for hardcoded workflows? `$name` could be enough? If so, any format for it?

Reply:
Honestly, then it's still a bit TBD depending on how we end up storing attributes for "internal workflows". E.g. if we ever added something like the example below.

```php
add_filter(
	'jpcrm_automation_workflows',
	function ( $workflows ) {
		// Contact Created: Force contact status to be "New".
		$workflow = new \Automattic\Jetpack\CRM\Automation\Automation_Workflow(
			[
				'id'           => 'testing',
				'name'         => __( 'New contact: Set status to "Customer"', 'zero-bs-crm' ),
				'description'  => __( 'This automation will change the status of a contact to "Customer" when they are created.', 'zero-bs-crm' ),
				'category'     => 'contact',
				'triggers'     => [
					\Automattic\Jetpack\CRM\Automation\Triggers\Contact_New::get_slug(),
				],
				'initial_step' => [
					'slug'            => 'contact_status_condition',
					'class_name'      => \Automattic\Jetpack\CRM\Automation\Actions\Update_Contact_Status::class,
					'attributes'      => [
						'new_status' => 'Customer',
					],
					'next_step_true'  => null,
					'next_step_false' => null,
				],
			]
		);

		$workflows[ $workflow->get_id() ] = $workflow;

		return $workflows;
	}
);
```

In this case, we'd have to rely on the manually defined ID to identify which workflow we're potentially updating attributes for via our APIs - and we cannot make it a number since it doesn't actually exist in the database.

We could end up storing everything in the database and in that case we'd be able to be strict about